src/util/domains.py

```diff
@@ -284,20 +284,59 @@ def clean_term(term):
-
-    {force_adapt} will run the adapter on all leaves having the removed field in the path. Useful
-    when deleting a field (in which case {new} is ignored).
+    Replace `old` by `new` in all domains of `model` and all its inheriting models using
+    an `adapter` callback.
+
+    `adapter` is a callback function to adapt leaves. Adapter functions must take three
+    arguments and return a domain that substitutes the original leaf. The arguments are:
+
+    - `leaf`: a domain leaf which is a `tuple` of the form `(left, op, right)`
+    - `in_or`: a boolean, when `True` it means the leaf is part of an OR (`"|"`) domain,
+      otherwise it is part of an AND (`"&"`) domain
+    - negated: a boolean, when `True` it means that the leaf is negated (`"!"`)
+
+    .. example::
+        .. code-block:: python
+
+            def adapter(leaf, in_or, negated):
+                left, op, right = leaf
+                ok, ko = (1, 2) if not negated else (2, 1)
+                if op == "="
+                    return [(left, "=", ok)]
+                elif op == "!=":
+                    return [(left, "=", ko)]
+                return [leaf]
+
+    `adapter` is called only on leafs that use the `old` field of `model` as **last** part
+    of the `left` part of leaves, unless `force_adapt` is `True. The domains returned by
+    an adapter do not need to have the `old` field replaced by `new` in the `left` part of
+    leaves. The replace will be done anyway to the whole domain returned by the adapter.
+    The purpose of the `adapter` is to modify the operator and the `right` part of the
+    leaves.
+
+    .. example::
+       When replacing `"field1"` by `"field2"`, the following happens:
+
+       - `("foo.bar.baz.field1", "=", 1)` gets adapted *only* if the record pointed to by
+         `foo.bar.baz` is of the requested `model`.
+       - `("foo.field1.baz", "=", 1)` is *not* adapted *even* if `foo` points to `model`,
+         unless `force_adapt` is `True`, because `field1` is not the last part of `left`
+         in this leaf.
+
+    .. note::
+       This function will replace domains in all *standard* domain fields. Including
+       filters, dashboards, and standard fields known to represent a domain.
+
+    :param str model: name of the model for which to adapt the domains
+    :param str old: name of the field to be adapted
+    :param str new: name of the field that should replace `old`
+    :param function adapter: adapter for leaves
+    :param list(str) skip_inherit: list of inheriting model names to don't adapt (skip)
+    :param bool force_adapt: when `True, run the `adapter` on all leaves having `new` in
+                                   `left` part of the leaf (path), useful when deleting a
+                                   field (in which case `new` is ignored).
+    :return: a `domain` that substitutes the input `leaf`
+    :rtype: :ref:`reference/orm/domains`
```

Comment:
Not true.
```suggestion
```

Reply:
Well what's the "cannonical" reference to _what_ is a domain?

Reply:
Ah never mind, this was meant to refer to the return of the `adapter` 🤦🏼

Reply:
Now I put the reference in the right part :)